Meridian59/Data/Models/AdminData/AdminData.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            TrackedObjects.Add(adminObject);
+            OnTrackedAdminObjectAdded(new AddTrackedAdminObjectEventHandlerArgs(this, adminObject));
+        }
+        public void UntrackAdminObject(AdminObject adminObject)
+        {
+            TrackedObjects.Remove(adminObject);
+        }
+
+        public AdminData()
+        {
+            TrackedObjects = new BaseList<AdminObject>();
+        }
+
+        /// <summary>
+        /// Interface to the Client's Network
+        /// </summary>
+        public event GameMessageEventHandler PacketSend;
+
+        #region events
+        //Event to add an object 
+        public delegate void AddTrackedAdminObjectEventHandler(object sender, AddTrackedAdminObjectEventHandlerArgs args);
+        public event AddTrackedAdminObjectEventHandler AdminObjectAdded;
+
+        protected void OnTrackedAdminObjectAdded(AddTrackedAdminObjectEventHandlerArgs e)
+        {
+            if (AdminObjectAdded != null) AdminObjectAdded(this, e);
+        }
+
+        //Event to pass a message back to the console
+        public delegate void LogAdminMessageEventHandler(LogAdminMessageEventHandlerArgs e);
+        public event LogAdminMessageEventHandler LogAdminMessage;
+        protected void OnLogAdminMessage(LogAdminMessageEventHandlerArgs e)
+        {
+            if (LogAdminMessage != null)
+            {
+                LogAdminMessage(e);
+            }
+        }
+        #endregion
+
+        void adminObjectProperty_ValueChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (sender.GetType() == typeof (AdminObjectProperty))
+            {
+                AdminObjectProperty changedProperty = (AdminObjectProperty)sender;
+                AdminObject changedAdminObject = changedProperty.GetOwner();
+
+                if (PacketSend != null)
+                {
+                    PacketSend(this,new GameMessageEventArgs(new ReqAdminMessage(string.Format("set object {0} {1} {2} {3}",
+                        changedAdminObject.ObjectNumber, changedProperty.PropertyName, changedProperty.PropertyType, changedProperty.PropertyValue))));
+                    PacketSend(this, new GameMessageEventArgs(new ReqAdminMessage(String.Format("show object {0}", changedAdminObject.ObjectNumber))));
+                }
+
+            }
+            else
+            {
+                throw new Exception("What?");
+            }
+        }
+
+        #region INotifyPropertyChanged
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        protected void RaisePropertyChanged(PropertyChangedEventArgs e)
+        {
+            if (PropertyChanged != null) PropertyChanged(this, e);
+        }
+
+        #endregion
+
+        #region IClearable
+        public void Clear(bool RaiseChangedEvent)
+        {
+            TrackedObjects.Clear();
+        }
+        #endregion
+
+        public void HandleAdminMessage(AdminMessage message)
```

Comment:
ReSharper keeps yelling at me about your standard haha. I need to set the options for this solution :)